cmake-init/templates/common/BUILDING.md

```diff
@@ -4,6 +4,17 @@
 
 For a list of dependencies, please refer to {% if vcpkg %}[vcpkg.json](vcpkg.json){% else %}[conanfile.py](conanfile.py){% end %}.{% end %}
 
+{% if pm_name=="conan" %}
+### Installing Conan Dependencies
+From the project root, run the following command:
+
+```sh
+conan install . -sbuild_type=<BUILD_TYPE> --build=missing
+```
+
+Where `<BUILD_TYPE>` is either `Release` or `Debug`.
+{% end %}
+
```

Comment:
The problem with this is twofold:

This adds content the the **user facing** documentation that is unnecessary. The CMake files are generated in a way that users are not required to use Conan. They could use anything else as well to provide the dependencies.

This add a Conan specific blurb that I think makes too much of an assumption. For someone who already knows how to use Conan, this is just noise. For someone new to Conan, they are better off reading Conan's own docs and learn that tool.

As a nitpick, I also don't like the conditional and the different command line style compared to HACKING.md.

Reply:
> This adds content the the **user facing** documentation that is unnecessary. The CMake files are generated in a way that users are not required to use Conan. They could use anything else as well to provide the dependencies.

As far as I know, there were not changes to the CMake (aside from presets) that would show up unless you select conan in the setup. 

> This add a Conan specific blurb that I think makes too much of an assumption. For someone who already knows how to use Conan, this is just noise. For someone new to Conan, they are better off reading Conan's own docs and learn that tool.

Which assumptions? You can either link Release or Debug libs from conan (also RelWithDebInfo, but that's compatible with Release afaik), hence the directory structure. For someone who knows how to use conan, they would have to change the presets to support both builds in a similar way. For someone who doesn't use Conan and wants to support both Deb and Rel, they will also have to figure out how to achieve this in a similar way.

I definitely see the benefits of having this automatically done by this tool, but as I mentioned, feel free to cancel PR if you don't feel the same.